### PR TITLE
Trust svg

### PR DESCRIPTION
**Pull request by pygy**
Add support for SVG in `m.trust()` strings, and some `DOMParser` support in the mocks.

## Description

This leverages the `DOMParser` API (IE 10+) to enable users to use `m.trust()` to create raw SVG fragments.

## How Has This Been Tested?
I've added corresponding tests for `render` and the mocks.

The [mocks test](https://flems.io/#0=N4IgZglgNgpgziAXAbVAOwIYFsZJAOgAsAXLKEAGhAGMB7NYmBvEAXwvW10QICsEqdBk2J4hcYgAIAJAHcIaACa1ZkgLyT5SlQB00eyZNr44ABxjUAFDpAARAPIBZAAoYATnBhvJYWt4DKAGoA4jYUPgCuaNTEEPSWAJTAejrEqQBu7jIOLu6ebilpxMYARjC+bjAAohjUhJZgUTFxaImSyfpFqdI5rh5e6jJayrL4vXlehamsCVPF1iAlGHAQ1HBhkdGx8UlzqbSWxACe5rRg2U59+Qn4MACOERhQcAuNWy02s-qdqRlZphNvBo0DBVD1LoDEikfkUDgD+t4FBIMNEYGcLrkETd7o9noc3BEYF9frDDic0ed4fl8FSYAAxNy0LD+YhuBQAc2xDyeLxsb2a9E+cxmcwONhgWFMx0kymoERwDA2-O2rQS7T2xEy3lpQMkILB4wRUJhf28ssGOppEwZTJZbLQ7IWAB44Ol2ZIAB5kNBwNQ6f0gEjEUyIAD0odkkfwsgAzPg-OzQwAmAAMadDrvZ-psAD4nRm3TmNjYIFgMOyYAX2QBqL1QIXfEn7Mmnc6y-Cy+UiKqwBXEG5oWjEKrcvFRRTlBQwRTErrzdudvs9iUifCDicAOS4XNxvJAmYbTfntGoHZPXYYy77a64ZlqMAAqgAlACSO55CyDIfDkdGsfjbiJqm6YHiAs4kgcC7nkuvarnU0CKButATnA+CwA6xCEO+eIpuBxAijCYogIwHpSOu8BKk0KptB0R5apIOqDPqGJXF4xp0Vk5oaJatI2syrIctYaCGCJIk2C6bqet6vrZoGxDBmGEZRv+CbJmmKZVrJRYgAYol6dWkjiSRxA5oQMBQFAtD5sZ2m6XpIkGUZMCkTmkjWc5Jk2HZ9mSI5IBOjZsh+FAijuS5XnCT5fn5pm2kUN5oklmWFZVrWZARfZxImsep6Lt2sEMGuyEwFuODYXuoF4c2UFyjBK6FZgOB3tQj6vuVn7yd+Sl-nGqnARplXQhx3jkXAgw1Rew4FcQ+DwSFSEoRqByjWhTDsph7UxlV8yjcgKYALpFZu263KOe7GYec7LcVcB7YdjXwACLXPm+p27h1Ck-spvWAWpIFupdEGWLtB2zYQCELfAq0YVhb0fgAjNt10oXdYMQzdqPkaVRJw3iNgAMQXWBS3AxjoNzYhZP3cVgRPIS7U2GZFm0IDpK7fD1PHWVuPnR5rPNuz923k9rWvTiH6M51im-tGP1AepmnE9lyPwMgHNo-NN3Q+tsPi3iiMk4LGuUyjoNYydet7oTfNK0eKu3erFOQ7dZs03TOOWwshm21dpMo0mnMlRbZ0LETSN+6rAc3k1IsvQzcmfd1ssAfL-1Zj7QO7VHTta+hOvtQbysR7d2fg5rpuB9j8fW6R-M7RjpfoxXR0wLTUD0zzCxBW4IWswR0xZdCxgEqqbAcCAD14LNcACDQ9CMMwPBsPtVBQAoADWCAoJwOB4LQZgWJQIARD3eBfnAilRKY6-srNTKhlgECYWyUAAALw-gH8AGyhvv5jUL-A+1Aj7HHMHgfgy9WBAA) also pass in Firefox, Chrome, Safari IE11 and Edge (thanks to @Yatekii for the last two).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated `docs/change-log.md`

### CODE REVIEW

**Comment by pygy** [comment]
I'll also add a SVG example in the docs

**Comment by pygy** [comment]
I'd like to rely on the same API for HTML, tough we'd probably have to adapt the `possibleParent` logic.

MathML can probably be done the same way, but I haven't tried.

**Comment by tivac** [comment]
This lgtm, but I know little enough about the underlying pieces that I'm not gonna approve the review. @isiahmeadows is probably a better choice.

**Comment by pygy** [comment]
@tivac I don't know much either. 

I knew there was a client-side markup parser, and I learnt how to use it yesterday.

There may be corner cases that are not handled properly, similar to the `possibleParent` table for HTML. I don't actually know what would break if we always used a `div` as a container for setting the `innerHTML`...

As mentioned in Gitter, I've tried to abuse synchronous XHR with data:URI to parse SVG in IE9, but it is rejected (using `XMLHttpRequest`, `XDomainRequest` and `ActiveXObject("MSXML2.XMLHTTP")`) but the trick works in browsers that don't need it since they supprt `DOMParser`.

File: render/render.js [hunk 1]
```diff
@@ -64,12 +71,16 @@ module.exports = function($window) {
 		insertNode(parent, vnode.dom, nextSibling)
 		return vnode.dom
 	}
-	function createHTML(parent, vnode, nextSibling) {
+	var possibleParents = {caption: "table", thead: "table", tbody: "table", tfoot: "table", tr: "tbody", th: "tr", td: "tr", colgroup: "table", col: "colgroup"}
```
Comment (isiahmeadows) [inline render/render.js]: I appreciate pulling this out. (I almost did over half a dozen times, and just didn't.)

**Comment by pygy** [comment]
Actually, please don't merge this yet, it can be done without the DOMParser API after all:

`temp = innerHTML = "<svg>" + value + "</svg>"; temp = temp.firstChild`

I'll update this tomorrow evening.

**Comment by pygy** [comment]
Here we go, less code, and IE9 compat (for the 1 out of 500 people who still use it worldwide 🙄 )

**Pull request merged**